### ISSUE TITLE
addon-a11y in config crashes with non-existent addons channel

### ISSUE THREAD

**Issue by donifer** [storybooks/storybook#6185]
**Describe the bug**
I'm trying to setup addon-a11y, however it crashes if I use the `addDecorator` in my `config.js`

Eg:
```js
import { withA11y } from '@storybook/addon-a11y';
import { withKnobs } from '@storybook/addon-knobs';
import { addDecorator, configure } from '@storybook/react';
import React from 'react';

const req = require.context('../src', true, /\.story\.tsx$/);

function loadStories() {
  req.keys().forEach(filename => req(filename));
}


addDecorator(withKnobs);
addDecorator(withA11y);

configure(loadStories, module);
```

Which crashes with:
```
bootstrap:781 Uncaught Error: Accessing non-existent addons channel, see https://storybook.js.org/basics/faq/#why-is-there-no-addons-channel
    at AddonStore.getChannel (index.js:30)
    at Object.<anonymous> (index.js:46)
    at Object.../../node_modules/@storybook/addon-a11y/dist/index.js
```

Seems to me that the channel is not ready yet [here](https://github.com/storybooks/storybook/blob/7251eac232a23408308959f857e06dc9be881955/addons/a11y/src/index.ts#L9).

However if I move the `addDecorator(withA11y);` to any story for example to a `SampleComponent.story.tsx` everything works as expected. However I really want to configure it in my `config.js`. 

What am I doing wrong?

**To Reproduce**
Use `addDecorator(withA11y)` in config.js

**Expected behavior**
To not crash

**System:**
 - OS: MacOS
 - Device: Macbook Pro 2016
 - Browser: Chrome
 - Framework: React
 - Addons: a11y
 - Version: 5.0.3

**Comment by vidal7** [comment]
For now, I found a workaround. Just export a function that return a story with the addDecorator
I'm in Vue and TypeScript but it should works fine in javascript if you remove the typings.

```ts
export function storiesOfComposant(nomComposant: Composants, module: NodeModule, filename: string): Story {
    const hierarchyRoot: string = getHierarchyRoot(nomComposant, filename);
    const espaceImportantePourVoirToutALaFin: string = ' ';

    const story: Story = storiesOf(hierarchyRoot + hierarchyRootSeparator + espaceImportantePourVoirToutALaFin + nomComposant, module);
    story.addDecorator(withA11y as any);

    return story;
}
```

**Comment by shilman** [comment]
We're using `withA11y` globally in `examples/official-storybook` inside the repo. Not sure what the difference is, but maybe you can look at that example to see if you can find any differences.

**Comment by deini** [comment]
I'm having the same issue. Seems to be an issue with the order of imports 🤔 

This doesn't work for me:
```js
import { withA11y } from '@storybook/addon-a11y';
import { addDecorator, configure } from '@storybook/react';
```

However, this works:
```js
import { addDecorator, configure } from '@storybook/react';
import { withA11y } from '@storybook/addon-a11y';
```

I'm able to reproduce it with `examples/official-storybook`.

**Comment by shilman** [comment]
That's pretty weird. Anybody of my `a11y` peeps wanna track this down? @CodeByAlex @jsomsanith @Armanio

**Comment by jsomsanith** [comment]
Quick investigation only : 
* import of `addDecorator` from `@storybook/react` executes `app/react/src/client/preview/index.js` that calls `@storybook/core/client` > `start()` that sets the channel. That's why it works when it is imported before a11y.
* import of anything from  `@storybook/addon-a11y` executes a `addons.getChannel()` that has not been set yet. It is used to set a listener to execute an aXe run on `EVENTS.REQUEST`.

Quick fix is what is suggested by @deini, but yes, need to find a way to register the listener once the channel is available.

**Comment by shilman** [comment]
@jsomsanith once again you do not disappoint. 😄 Thank you kindly!

**Comment by Armanio** [comment]
Intresting. I'll try fix it in as soon as possible.

**Comment by Armanio** [comment]
Hi guys! Sorry for my long absence. I create PR, check it. ⬆️

**Comment by trippingtarballs** [comment]
Not sure if this fix has worked. I'm using `v5.0.6` and having the same issue.

**Comment by shilman** [comment]
@trippingtarballs the change hasn't been released yet

**Comment by shilman** [comment]
w00t!! I just released https://github.com/storybooks/storybook/releases/tag/v5.1.0-alpha.21 containing PR #6354 that references this issue. Upgrade today to try it out!

Because it's a pre-release you can find it on the `@next` NPM tag.

Closing this issue. Please re-open if you think there's still more to do.
